packages/components/calendar/src/use-calendar-picker.ts

```diff
@@ -221,11 +177,30 @@ export function useCalendarPicker(props: CalendarPickerProps) {
 
       const nextItem = map.get(nextValue);
 
+      scrollTo(nextValue, list);
```

Comment:
Two things need fixing in the Home/End key behavior:
1. The year isn't changing.
2. It's jumping straight to the start/end instead of moving smoothly as it did before.